src/twisted/plugins/cred_unix.py

```diff
@@ -39,11 +38,14 @@ def verifyCryptedPassword(crypted, pw):
 
     if crypt is None:
         raise NotImplementedError("cred_unix not supported on this platform")
-    if not isinstance(pw, StringType):
+    if isinstance(pw, bytes):
         pw = pw.decode('utf-8')
-    if not isinstance(crypted, StringType):
+    if isinstance(crypted, bytes):
         crypted = crypted.decode('utf-8')
-    return crypt.crypt(pw, crypted) == crypted
+    try:
+        return crypt.crypt(pw, crypted) == crypted
```

Comment:
@mgorny In what case does `crypt.crypt()` throw an `OSError`?

In this PR: https://github.com/ansible/ansible/pull/70246

it mentions that `crypt.crypt` in Python 3.9 will throw an OSError if the algorithm is not supported.  Is there a test that we can write to trigger this error, so that there is test coverage of the `except OSError:` branch?

Reply:
Looks like `crypt.crypt()` throws an OSError if the underlying `crypt()` C function fails due to this change in Python 3.9:

https://github.com/python/cpython/commit/0d3fe8ae4961bf551e7d5e42559e2ede1a08fd7c

It might not be so easy to write a test for this using the existing Python `crypt.crypt` function.
Using a mock of crypt which always throws an `OSError` as @glyph suggested might be the easier way to go.

Reply:
That's weird. The current set of tests was sufficient to get it to throw for me. IIRC it was throwing on one of the invalid password cases.

Reply:
Ah, maybe it's because `crypt.crypt` only throws `OSError` on Python 3.9, but none of the CI for Twisted runs on Python 3.9.
In that case it would be good to write a test which mocks out `crypt.crypt` and always throws OSError.

Reply:
```python
In [1]: import crypt                                                                                                                               

In [2]: crypt.crypt('test', '')                                                                                                                    
---------------------------------------------------------------------------
OSError                                   Traceback (most recent call last)
<ipython-input-2-7185f9de0941> in <module>
----> 1 crypt.crypt('test', '')

/usr/lib/python3.9/crypt.py in crypt(word, salt)
     80     if salt is None or isinstance(salt, _Method):
     81         salt = mksalt(salt)
---> 82     return _crypt.crypt(word, salt)
     83 
     84 

OSError: [Errno 22] Invalid argument
```

Reply:
Yes, but that only fails like that on Python 3.9, and Twisted's CI does not yet run on Python 3.9

Reply:
I'm sorry, I've written my comment before reading yours ;-). I'm already working on a mocked test case, just need a few minutes to run the suite.

Reply:
No worries. :)
It looks like Travis also has a **3.9-dev** target, see: https://github.com/jazzband/pip-tools/pull/1126/files
.

If you want, we can also add **3.9-dev** to .travis.yml .
That will be another way to get the coverage numbers up.

Reply:
I've pushed a test that should cover the coverage problem. Ofc, enabling tests on 3.9 would also be nice.

Reply:
Thanks, this looks ok now, and coverage is better: https://coveralls.io/builds/32572286